py_ballisticcalc/conditions.py

```diff
@@ -5,88 +5,159 @@
+cPressureExponent: float = 5.255876     # =g*M/R*L
 cA0: float = 1.24871
 cA1: float = 0.0988438
 cA2: float = 0.00152907
 cA3: float = -3.07031e-06
 cA4: float = 4.21329e-07
 cA5: float = 3.342e-04
-cStandardTemperature: float = 59.0
-cStandardPressure: float = 29.92
-cStandardDensity: float = 0.076474
-
-cIcaoTemperatureDeltaR: float = cIcaoStandardTemperatureR - cIcaoFreezingPointTemperatureR
-
+# ISA, metric units: (https://www.engineeringtoolbox.com/international-standard-atmosphere-d_985.html)
+cDegreesCtoK: float = 273.15              # °K = °C + 273.15
+cStandardTemperatureC: float = 15.0       # °C
+cLapseRateMetric: float = -6.5e-03        # Lapse Rate, °C/m
+cStandardPressureMetric: float = 1013.25  # hPa
+cSpeedOfSoundMetric: float = 331.3        # Mach1 in m/s = cSpeedOfSound * sqrt(°K)
+cStandardDensityMetric: float = 1.2250    # kg/m^3
+cDensityImperialToMetric: float = 16.0185 # lb/ft^3 to kg/m^3
+# ICAO standard atmosphere:
+cDegreesFtoR: float = 459.67              # °R = °F + 459.67
+cStandardTemperatureF: float = 59.0       # °F
+cLapseRateImperial: float = -3.56616e-03  # Lapse rate, °F/ft
+cStandardPressure: float = 29.92          # InHg
+cSpeedOfSoundImperial: float = 49.0223    # Mach1 in fps = cSpeedOfSound * sqrt(°R)
+cStandardDensity: float = 0.076474        # lb/ft^3
 
 @dataclass
 class Atmo(TypedUnits):  # pylint: disable=too-many-instance-attributes
-    """Stores atmosphere data for the trajectory calculation"""
-
+    """Atmospheric conditions and density calculations"""
     altitude: [float, Distance] = field(default_factory=lambda: Set.Units.distance)
     pressure: [float, Pressure] = field(default_factory=lambda: Set.Units.pressure)
     temperature: [float, Temperature] = field(default_factory=lambda: Set.Units.temperature)
-    humidity: float = 0.78
-    density: float = field(init=False)
-    mach: Velocity = field(init=False)
-    _mach1: Velocity = field(init=False)
-    _a0: float = field(init=False)
-    _t0: float = field(init=False)
-    _p0: float = field(init=False)
-    _ta: float = field(init=False)
+    humidity: float = 0.0           # Relative humidity [0% to 100%]
+    density_ratio: float = field(init=False)  # Density / cStandardDensity
+    mach: Velocity = field(init=False)  # Mach 1 in reference atmosphere
+    _mach1: float = field(init=False)  # Mach 1 in reference atmosphere in fps
+    _a0: float = field(init=False)  # Initial reference altitude (ft)
+    _t0: float = field(init=False)  # Temperature given at reference altitude °F
+    _p0: float = field(init=False)  # Barometric pressure (sea level)
+    _ta: float = field(init=False)  # Standard temperature at reference altitude °F
 
     def __post_init__(self):
-
         if self.humidity > 1:
-            self.humidity = self.humidity / 100
+            self.humidity = self.humidity / 100.0
         if not 0 <= self.humidity <= 1:
-            self.humidity = 0.78
+            self.humidity = 0.0
```

Comment:
I don't understand how this would handle the desired functionality, which is:

- Valid humidity range is [0, 1]
- Default humidity is 0%
- If value is passed, it might be in "percent" terms [0, 100], in which case we divide by 100 to put it in float terms [0, 1].
- If even after that correction humidity is not in valid range [0, 1] then we apply default value.  **I suppose that better logic would be to throw exception in this case?**

Reply:
Ok, you're right